### ISSUE TITLE
Skip the shortwave solve for all-dark column chunks (compute win)

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#516]
## Summary

Radiation currently runs the **shortwave (SW) solve on every column**, including the ~half of the globe that is dark at any instant. Because the per-column gas-optics scheme is `vmap`'d, night columns can't be skipped *inside* the kernel (`vmap` is SIMD — a per-lane `cond` degrades to a `select` that runs both branches). The compute win has to be recovered at the **driver level**, by gating whole column chunks/shards on a **scalar** predicate before they enter the `vmap`.

This is the *compute* follow-up to the *memory* fix in jax-rrtmgp ([#8](https://github.com/climate-analytics-lab/jax-rrtmgp/issues/8) / PR [#9](https://github.com/climate-analytics-lab/jax-rrtmgp/pull/9)). That PR removed a `jax.lax.cond(zenith ≥ π/2)` inside `solve_sw` that was blowing up memory under the column `vmap` (~31 GB at T63L47). It deliberately replaced the night short-circuit with a compute-everything-then-mask approach — correct and memory-light, but night columns now pay full SW cost. This issue is about earning that cost back, and it **composes with** (depends on) that fix.

## Mechanism

Gate a whole chunk/shard of columns with a scalar reduction over its zeniths, *around* the `vmap`:

```python
def gated_chunk_sw(zen_chunk, *args):          # zen_chunk: [chunk_size]
    all_night = jnp.all(zen_chunk >= 0.5 * jnp.pi)        # SCALAR predicate
    return jax.lax.cond(
        all_night,
        lambda: zeros_like_sw_fluxes,
        lambda: jax.vmap(solve_sw_column)(zen_chunk, *args),
    )
```

Because the predicate is a scalar (not per-lane-batched), XLA keeps it a real runtime `conditional`: the dark chunk's SW solve is genuinely skipped, and — crucially — it does **not** reintroduce the per-column table broadcast from rrtmgp #8.

`solve_sw` already returns **exact zeros** for night columns (via the mask in PR #9), so a chunk straddling the terminator is still correct when it is *not* skipped.

## Evidence

Measured on the real `solve_sw` (full SW g224, 32-column chunk, CPU):

```
UNGATED  vmap(solve_sw):          GATED  lax.cond(all_night) around vmap:
  all-day   chunk:  28.2 ms         all-day   chunk:  27.5 ms  (unchanged)
  all-night chunk:  26.3 ms         all-night chunk:   0.0 ms  → ~700x faster
HLO: real runtime `conditional` (true skip), no per-column kmajor broadcast
```

## How much it's worth

- **If columns are chunked/sharded by geography** (contiguous tiles — the usual case): ~half the tiles are fully dark at any instant → roughly half of SW skips *for free, no sorting*. A night-side shard skips its entire SW solve.
- **If chunks interleave day/night**: few chunks are all-dark → little benefit unless columns are first **sorted by zenith** (a gather/scatter on the column axis, cheap vs RT) so dark columns form contiguous skippable chunks.
- **Finer chunks** → more fully-dark chunks skipped, but more `cond` overhead and lower `vmap` occupancy per chunk. There's a sweet spot.
- Ceiling: LW always runs and SW is ~half of radiation, with night fraction ~50% → realistic max ≈ **20–25% of total radiation** wall-clock (more if the config is SW-heavy). Columns straddling the terminator are the irreducible residue.

The alternative that hits the ~50%-of-SW ceiling regardless of geometry is **day-column compaction** (gather lit columns → solve a static-padded compact batch → scatter back). It eliminates terminator waste but adds gather/scatter + static-shape bookkeeping — more complexity for the last few percent.

## Proposed work

1. In the radiation driver, wrap the per-chunk/per-shard SW call in `lax.cond(jnp.all(zenith_chunk ≥ π/2), zeros, vmap(solve_sw))`. Confirm the predicate is a scalar tracer so it lowers to a `conditional`, not a `select`.
2. Check how columns map to chunks/shards today; if they're not geographically clustered, evaluate sorting columns by solar zenith before chunking.
3. (Optional, later) day-column compaction if the terminator residue shows up in profiles.

### Acceptance
- All-dark chunks skip the SW solve (verify via HLO `conditional` / a flop or wall-clock drop).
- No per-column gas-optics table broadcast is reintroduced (the rrtmgp #8 symptom).
- Heating rates unchanged vs the ungated path (night columns already return zeros).

cc: depends on jax-rrtmgp PR [#9](https://github.com/climate-analytics-lab/jax-rrtmgp/pull/9).

**Comment by duncanwp** [comment]
Cancelled - the gather/scatter overhead made this more expensive than just running rrtmgp over all the columns (after teh per-column speedup was merged)
